### PR TITLE
Allow anonymous access in "CurlClient" class

### PR DESCRIPTION
**Pull request by digitalkaoz**
public jiras allow anonymous access...so there is no need to disallow it here

### CODE REVIEW

File: src/Jira/Api/Client/CurlClient.php [hunk 1]
```diff
@@ -52,10 +52,6 @@ public function __construct()
      */
     public function sendRequest($method, $url, $data = array(), $endpoint, AuthenticationInterface $credential, $isFile = false, $debug = false)
     {
-        if (!($credential instanceof Basic)) {
```
Comment (aik099) [inline src/Jira/Api/Client/CurlClient.php]: Instead of removing use approach taken in `PHPClient`:

```php
if (!($credential instanceof Basic) && !($credential instanceof Anonymous)) {
    throw new \Exception(sprintf("PHPClient does not support %s authentication.", get_class($credential)));
}
```

File: src/Jira/Api/Client/CurlClient.php [hunk 2]
```diff
@@ -65,7 +61,9 @@ public function sendRequest($method, $url, $data = array(), $endpoint, Authentic
         curl_setopt($curl, CURLOPT_URL, $endpoint . $url);
         curl_setopt($curl, CURLOPT_HEADER, 0);
         curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
-        curl_setopt($curl, CURLOPT_USERPWD, sprintf("%s:%s", $credential->getId(), $credential->getPassword()));
+        if ($credential instanceof Basic) {
```
Comment (aik099) [inline src/Jira/Api/Client/CurlClient.php]: Please replace with `if (!($credential instanceof Anonymous)) {` (approach from `PHPClient` class) to allow custom authentication ways to be used.

**Comment by digitalkaoz** [comment]
done

File: src/Jira/Api/Client/CurlClient.php [hunk 3]
```diff
@@ -52,10 +53,10 @@ public function __construct()
      */
     public function sendRequest($method, $url, $data = array(), $endpoint, AuthenticationInterface $credential, $isFile = false, $debug = false)
     {
-        if (!($credential instanceof Basic)) {
-            throw new \Exception(sprintf("CurlClient does not support %s authentication.", get_class($credential)));
+        if (!($credential instanceof Basic) && !($credential instanceof Anonymous)) {
+            throw new \Exception(sprintf("PHPClient does not support %s authentication.", get_class($credential)));
```
Comment (aik099) [inline src/Jira/Api/Client/CurlClient.php]: I haven't noticed, that exception message was containing class name. Please change `PHPClient` back to `CurlClient`.

File: src/Jira/Api/Client/CurlClient.php [hunk 4]
```diff
@@ -52,10 +53,10 @@ public function __construct()
...
         }
-
+
```
Comment (aik099) [inline src/Jira/Api/Client/CurlClient.php]: Please revert this change. Also enabling setting in IDE to trim whitespaces on empty lines is a recommended approach.

Comment (digitalkaoz) [inline src/Jira/Api/Client/CurlClient.php]: did it completel in the github ui, so sorry

Comment (aik099) [inline src/Jira/Api/Client/CurlClient.php]: It really makes contributions easier indeed. Don't worry, we'll be adding automated coding standard checks later on and then you can see check result right no Scrutinizer CI website.

**Comment by aik099** [comment]
Looks good. Please also rebase.

**Comment by digitalkaoz** [comment]
so all done @aik099 also squashed everything

**Comment by aik099** [comment]
Looks good. Thank you @digitalkaoz , merging.

**Comment by digitalkaoz** [comment]
your welcome :)
